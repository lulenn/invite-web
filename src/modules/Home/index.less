.home {
  height: calc(100vh - 160px);
  position: relative;
}

.content {
  overflow: auto;
  text-align: center;
  position: relative;
  top: 50%;
  transform: translateY(-50%);

  p {
    margin: 0 0 24px 0;

    &.text-main {
      font-size: 36px;
    }

    &.text-hint {
      font-size: 16px;
    }
  }
}
