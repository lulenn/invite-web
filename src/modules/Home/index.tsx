import { useState, useCallback } from 'react';
import { Button } from 'antd';
import UserModal from '../../components/UserModal';
import './index.less';
import SuccessModal from '../../components/SuccessModal';

function Home(): JSX.Element {
  const [userModalVisible, setUserModalVisible] = useState(false);
  const [successModalVisible, setSuccessModalVisible] = useState(false);

  const openUserModal = useCallback(() => {
    setUserModalVisible(true);
  }, []);
  const handleOk = useCallback(() => {
    setUserModalVisible(false);
    setSuccessModalVisible(true);
  }, []);
  const handleCancel = useCallback(() => {
    setUserModalVisible(false);
  }, []);
  const handleSuccess = useCallback(() => {
    setSuccessModalVisible(false);
  }, []);

  return (
    <div className="Home">
      <div className="Content">
        <p className="text-main">
          A better way <br /> to enjoy every day.
        </p>
        <p className="text-hint">Be the first to know when we launch.</p>
        <Button onClick={openUserModal}>Request an invite</Button>
      </div>

      <UserModal data-testid="user-madal" visible={userModalVisible} onOk={handleOk} onCancel={handleCancel} />
      <SuccessModal data-testid="success-madal" visible={successModalVisible} onOk={handleSuccess} />
    </div>
  );
}

export default Home;
