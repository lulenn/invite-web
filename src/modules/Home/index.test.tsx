import { render } from '@testing-library/react';
import Home from '.';

describe('Home', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });
});
