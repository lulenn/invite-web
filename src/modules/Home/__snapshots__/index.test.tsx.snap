// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="home"
  >
    <div
      class="content"
    >
      <p
        class="text-main"
      >
        A better way 
        <br />
         to enjoy every day.
      </p>
      <p
        class="text-hint"
      >
        Be the first to know when we launch.
      </p>
      <button
        class="ant-btn"
        type="button"
      >
        <span>
          Request an invite
        </span>
      </button>
    </div>
    <div
      class="ant-modal-root"
    >
      <div
        class="ant-modal-wrap"
        role="dialog"
        style="display: none;"
        tabindex="-1"
      />
    </div>
  </div>
</DocumentFragment>
`;
